### DISCUSSION TITLE
AdjustContrastd & ShiftIntensityd explanation

### DISCUSSION THREAD

**Discussion by hreso110100** [Project-MONAI/MONAI#6461] [Q&A] [answered]
Hi as far as I know ScaleIntensityd can rescale HU units to specific range right ? But it means that for example when I apply ScaleIntensityd with range -200 to 200 HU then I can adjust contrast only within this range with AdjustContrastd/ShiftIntensityd ? Or these transforms operate on pixel level values ranged from 0 to 255 ? I am confused...

**Comment by KumoLiu** [comment, marked as answer]
Hi @hreso110100, if you want to apply scaling from [a_min, a_max] to [b_min, b_max] with the clip option. I would suggest you use `ScaleIntensityRange` instead of `ScaleIntensity`. `ScaleIntensityd` means scale the whole image array to [minv, maxv] in pixel level, the output will be in this range.

Hope it can help you, thanks!

**Reply by hreso110100** [reply]
Hi @KumoLiu so if I use ScaleIntensityRange with this values for example   (a_min=0, a_max=2000, b_min=0.0, b_max=1.0) the my data in pixel level will between 0 and 1 ? Right ? ...then other transforms like AdjustContrastd/ShiftIntensityd also operate on values between 0 and 1 or it operates on HU values from 0 to 2000?

**Reply by KumoLiu** [reply]
Hi @hreso110100, after you scaled, the following transforms will be operated on values between 0-1. Thanks!
